### PR TITLE
feat(forms): Adding urgency category and related question type

### PR DESCRIPTION
**Pull request by ccailly**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 

Added `Urgency` category and related sub-question type : `Urgency`

image
image

### CODE REVIEW

File: src/Form/QuestionType/QuestionTypeUrgency.php [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+namespace Glpi\Form\QuestionType;
+
+use CommonITILObject;
+use Glpi\Application\View\TemplateRenderer;
+use Glpi\Form\Question;
+use Override;
+
+/**
+ * Long answers are multiple lines inputs used to answer questions with as much details as needed.
+ */
```
Comment (AdrienClairembault) [inline src/Form/QuestionType/QuestionTypeUrgency.php:45]:
```suggestion
```

**Review by AdrienClairembault — CHANGES_REQUESTED** [review]

**Comment by AdrienClairembault** [comment]
Functional test is OK, only one thing.

Could you add an empty value ?
We are forced to set up a default value for now:

image

Maybe some users will want to leave the default value empty here.

**Review by AdrienClairembault — APPROVED** [review]

**Review by trasher — APPROVED** [review]

File: src/Html.php [hunk 2]
```diff
@@ -4788,106 +4788,22 @@ public static function jsAdaptDropdown($id, $params = [])
...
-               // and return the group object
-               if (filteredChildren.length) {
-                  var modifiedData = $.extend({}, data, true);
-                  modifiedData.children = filteredChildren;
+        $js = <<<JS
+            select2_configs['{$id}'] = {
+                type: 'adapt',
+                field_id: '{$id}',
+                width: '{$width}',
+                placeholder: '{$placeholder}',
```
Comment (cedric-anne) [inline src/Html.php:4796]: You have to adapt the `$placeholder = "placeholder: " . json_encode($params["placeholder"]) . ","` line and use the json encoded value here withour surrounding it by quotes.
```suggestion
                placeholder: {$placeholder},
```

File: src/Html.php [hunk 3]
```diff
@@ -4788,106 +4788,22 @@ public static function jsAdaptDropdown($id, $params = [])
...
+                templateresult: {$templateresult},
+                templateselection: {$templateselection}
+            };
+JS;
 
-                  // You can return modified objects from here
-                  // This includes matching the `children` how you want in nested data sets
-                  return modifiedData;
-               }
+        if ($params['init']) {
```
Comment (cedric-anne) [inline src/Html.php:4803]: This is a new parameter. Even if there is no usage of the `Html::jsAdaptDropdown()` outside of the `Dropdown::showFromArray()` method, you should ensure that calling this method without defining `$params['init']` will have the same behaviour as the previous code.

```suggestion
        if ($params['init'] ?? true) {
```

Also, you will have to explicitely pass `'init' => false` if you do not want this dropdown to be initialized.

**Review by cedric-anne — CHANGES_REQUESTED** [review]

File: src/Html.php [hunk 4]
```diff
@@ -4782,112 +4782,28 @@ public static function jsAdaptDropdown($id, $params = [])
 
         $placeholder = '';
         if (isset($params["placeholder"])) {
```
Comment (cedric-anne) [inline src/Html.php:4784]:
```suggestion
```

File: src/Html.php [hunk 5]
```diff
@@ -4782,112 +4782,28 @@ public static function jsAdaptDropdown($id, $params = [])
...
-            $placeholder = "placeholder: " . json_encode($params["placeholder"]) . ",";
+            $placeholder = json_encode($params["placeholder"]);
         }
```
Comment (cedric-anne) [inline src/Html.php:4786]:
```suggestion
```

File: src/Html.php [hunk 6]
```diff
@@ -4782,112 +4782,28 @@ public static function jsAdaptDropdown($id, $params = [])
 
         $placeholder = '';
         if (isset($params["placeholder"])) {
-            $placeholder = "placeholder: " . json_encode($params["placeholder"]) . ",";
+            $placeholder = json_encode($params["placeholder"]);
```
Comment (cedric-anne) [inline src/Html.php:4785]:
```suggestion
        $placeholder = json_encode($params["placeholder"] ?? '');
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**